optuna/integration/allennlp/_variables.py

```diff
@@ -51,23 +50,23 @@ def _get_key(self, name: str) -> str:
             raise KeyError(f"{name} is not found in `{self.NAME_OF_PATH}`.")
         return key
 
-    def set_value(self, name: str, value: str) -> None:
+    def set_value(self, name: str, value: Any) -> None:
         """Set values to environment variables.
 
         `set_value` is only invoked in `optuna.integration.allennlp.AllenNLPExecutor`.
 
         """
         key = self._get_key(name).format(self.target_pid)
-        os.environ[key] = value
+        os.environ[key] = json.dumps(value)
 
-    def get_value(self, name: str) -> Optional[str]:
+    def get_value(self, name: str) -> Any:
         """Fetch parameters from environment variables.
 
         `get_value` is only called in `optuna.integration.allennlp.AllenNLPPruningCallback`.
 
         """
         key = self._get_key(name).format(self.target_pid)
         value = os.environ.get(key)
-        assert value is not None, f"{key} is not found in environment variables."
-
-        return value
+        if value is None:
+            raise KeyError(f"{key} is not found in environment variables.")
```

Comment:
Since this `get_value` is only called inside the allennlp integration, we should assert the value is not None instead of rising an error.
```
assert value is not None
```

Reply:
Hmm, I think it conflicts with the previous discussion.
https://github.com/optuna/optuna/pull/2977#discussion_r756009719

> we should assert the value is not None instead of rising an error

`assert` statement also raises an error (`AssertionError`) so the difference is only the type of an exception (Both could raise an error). I'd like to know what you intend in more detail. 🐈

Reply:
I missed that comment. Thanks for the information. I think we need to explain all errors that users may face except `AssertionError` in the document, but do not the current one. But, users must not face this error, it's OK to leave as this is. Let me resolve this comment.